pulsar/consumer_impl.go

```diff
@@ -453,16 +465,16 @@ func (c *consumer) Ack(msg Message) error {
 
 // AckID the consumption of a single message, identified by its MessageID
 func (c *consumer) AckID(msgID MessageID) error {
-	mid, ok := c.messageID(msgID)
-	if !ok {
-		return errors.New("failed to convert trackingMessageID")
-	}
-
-	if mid.consumer != nil {
-		return mid.Ack()
+	partition := int(msgID.PartitionIdx())
```

Comment:
> Why not make a method to check the partition or the msgID?

Good suggestion, I will make the method.

> You can improve messageID() method.

`messageID()` converts the `MessageID` to `trackingMessageID`. It will cause the type lost of `chunkMessageID`.